### ISSUE TITLE
Request: Binary releases

### ISSUE THREAD

**Issue by m040601** [sp1ff/mpdpopm#2]
Very interesting project making use of the mpd stickers and channels.
Really  promising and well thought the use cases you describe in the README.
Good to know someone is building something in Rust or Go,  making use of this underrated and quite unknown feature of mpd.

I haven't yet tried out, but I'm eager to.

Would it be possible to release it as a binary ? At least compiled for x86_64 ?

It would make much easier for people who dont have / dont want to download the 500 mega rust distribution just to quickly try it out.

Also, think of low spec devices like less powered old laptops or the Rasbperry Pi, where constantly recompiling compiling may not be that desirable

Perhaps you can also have a look at how other cli rust tools do it, for example,

    skim, https://github.com/lotabout/skim/releases
    fd, https://github.com/sharkdp/fd

I assume you are aware of github actions right ?
Have a look at how ripgrep, https://github.com/BurntSushi/ripgrep does it, https://github.com/BurntSushi/ripgrep/tree/master/.github/workflows

Thanks in advance

**Comment by sp1ff** [comment]
It never occurred to me that someone would not have Rust installed :)
JK-- the truth is that while I'm aware of of github actions, I'm largely ignorant of what they can do for me. It's a great suggestion & I'm learning more about them right now. Will ping you back here when I've got something up & running. Thanks!

**Comment by sp1ff** [comment]
@m040601 -- binary releases for Linux & Mac up.

**Comment by m040601** [comment]
Thanks a lot !

I think you mispelled the binary files realesed. You are shipping two binary files in the folder "bin":

- mppopm
- mppopmd

I think you meant "mp**d**popm" and "mp**d**popmd" instead.

-

I'm already testing it and it seems to run at least.
Hope we can also create an AUR package for Archlinux based on the binary, so that more people can try it out easily. 


Just before you close this issue, and to avoid opening another one with simple questions.

1. Could you also update the README with a couple more of **practical examples**  running mpdpopm ? With their actual shell output ? So that one can follow along and see the expected output.

 It may sound "obvious" to you, because you designed the program, and you already have it very clear in your mind "where you wanna arrive" 

> 
Once up, it will create an mpd channel ... and begin monitoring mpd playback. ....it will .... in the sticker database.


Also , under "prerequisites", document, what mpd configuration lines must be correct.


> ...I am currently using mpdpopm day in & day out with my music collection, but it’s early days. ....

I know it's too early, but a lot of this usage of mpd stickers/channels is very poorly documented officially and I'm having a hard time "discovering" and troubleshooting how to use it, without practical real life examples. One thing I'm am sure is that this has great potential.

I'm also having this annoying

**Comment by sp1ff** [comment]
Thanks for the detailed feedback-- this is awesome. Let me respond to one thing at a time:

> I think you mispelled the binary files realesed.

Not misspelled, tho perhaps ill-considered. I intentionally named the daemon `mppopmd` and the corresponding cli `mppopm` (`mpdpopmd` just seemed ungainly).

> Hope we can also create an AUR package for Archlinux based on the binary, so that more people can try it out easily.

Yes, there are a couple of cargo add-ons for producing Debian & Arch packages that I plan to add to the release workflow.

> Could you also update the README...

Sure.

> Also , under "prerequisites", make sure to document, what mpd configuration lines must be there already...

So this is a little more interesting. I had _assumed_ that in the absence of a `sticker_file` line in `mpd.conf` mpd will just use a default, but now I'm not sure... OK, just perused the MPD source code. If you don't configure one, sticker commands will fail with an error message "sticker database is disabled". So-- I will indeed add that to the pre-requisites.
RE perms, make sure you have write access to the parent directory, as well. For reference, here are the perms for mine:
```
┌┤ /home/mgh (Sat Nov 28 12:18:08) - n/a msg(s), 1 usr(s), up 26 days, 21:55, load 2.49/6, avail. mem 8/23G
└──5000:mgh@Crickhollow[0,0,3]: ls -al var/lib/mpd
total 1164
drwxr-xr-x 2 mgh users   4096 Nov 19 08:28 .
drwxr-xr-x 4 mgh users   4096 Oct  5 12:21 ..
-rw-r--r-- 1 mgh users  39306 Nov 26 08:06 state
-rw-r--r-- 1 mgh users 770048 Nov 19 08:28 sticker.sql
-rw-r--r-- 1 mgh users 368640 Nov 15 14:57 sticker.sql.1
```
Do let me know if that solves it & I'll update the docs accordingly.

OK-- stickers: I agree that they're not documented super-well: [this](https://www.musicpd.org/doc/html/protocol.html#stickers) and the MPD source code are what I've been working off of. If you tell me more about what you want to accomplish, I can help. Right now, `mpdpopm` is the bare-bones of an app. It's plumbing, not the sink. You can use the `mppopm` CLI to, say, rate the current song, but in order to actually do anything with that rating in the future, you're going to have to write some kind of mpd client for yourself. For example, I have a shell script that queries the sticker database for all songs with no rating & queues 'em up for listening. Or, I have some Emacs lisp that queues up all songs with a rating > 128. What mpd client(s) do you use and how are you planning to make use of the information `mpdpopm` adds?

FWIW, my next feature is going to be a "filter" command that extends mpd filters to add support for the `mpdpopm` stickers. So, you could send a command to `mpdpopm` saying something like "queue up all tracks with a rating > three stars and with a date greater than 01/01/20", say.

**Comment by sp1ff** [comment]
README updated, binaries provided, along with a Debian binary package. Tried Arch (via both `cargo-arch` & `cargo-aur`) without success. Opened a separate issue for that, but wanted to push these changes out without waiting on getting that figured out.

**Comment by m040601** [comment]
Thanks for the updates.

Just wanted to report, that it seems to be working now. At least I can now play with:

1. Keeping an eye on the sqlite database, sticker.db.sql to see what this actions trigger exactly. Seems to be getting updated with name/value pairs:

`unwoundstack.com:rating` 

example a one star rating, gives a numerical value of 1, a five star rating gives a numerical value of 255

You'll have to document this, so that end users can understand exacty whats happening.



2. opening a shell and monitoring this:

`mpdpopmd -F -v log`

3. open another shell and play issuing:

`mppopm set-rating '*****'`


What is the exact syntax ? How many stars max/min ? what kind of quotes are needed (because of the shell) ? Etc.

Also needs documentation.

**Comment by sp1ff** [comment]
RE the README, agreed-- it's getting big, but my priority was to get the information _out_ of my head and _on_ to paper (so to speak). As far as re-factoring the content now that it's out there, I'm more inclined to move it into man pages or above all tool-specific documentation available via `--help`.

Thanks for all the suggestions!
